sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ConditionalSimplificationCodeFixProvider.cs

```diff
@@ -37,239 +37,250 @@ public sealed class ConditionalSimplificationCodeFixProvider : SonarCodeFixProvi
+                    }), context.Diagnostics);
             }
         }
 
-        private static CodeAction GetActionToExecute(CodeFixContext context, SyntaxNode root,
-            SyntaxNode nodeToChange, SyntaxNode nodeToAdd, SyntaxAnnotation annotation)
+        private static SyntaxNode Simplify(Diagnostic diagnostic, SemanticModel semanticModel, SyntaxNode oldNode, out SyntaxAnnotation annotation)
         {
-            return CodeAction.Create(
-                Title,
-                c =>
-                {
-                    var nodeToAddWithoutAnnotation = RemoveAnnotation(nodeToAdd, annotation);
-
-                    var newRoot = root.ReplaceNode(
-                        nodeToChange,
-                        nodeToAddWithoutAnnotation.WithTriviaFrom(nodeToChange).WithAdditionalAnnotations(Formatter.Annotation));
-                    return Task.FromResult(context.Document.WithSyntaxRoot(newRoot));
-                });
+            ExpressionSyntax compared;
+            switch (oldNode)
+            {
+                case ConditionalExpressionSyntax conditional:
+                    var condition = conditional.Condition.RemoveParentheses();
+                    var whenTrue = conditional.WhenTrue.RemoveParentheses();
+                    var whenFalse = conditional.WhenFalse.RemoveParentheses();
+                    ConditionalSimplification.TryGetExpressionComparedToNull(condition, out compared, out _);
+                    return SimplifyCoalesceExpression(new ComparedContext(diagnostic, semanticModel, compared, out annotation), whenTrue, whenFalse);
+
+                case IfStatementSyntax ifStatement:
+                    var ifPart = ConditionalSimplification.ExtractSingleStatement(ifStatement.Statement);
+                    var elsePart = ConditionalSimplification.ExtractSingleStatement(ifStatement.Else?.Statement);
+                    ConditionalSimplification.TryGetExpressionComparedToNull(ifStatement.Condition, out compared, out var _);
+                    return SimplifyIfStatement(new ComparedContext(diagnostic, semanticModel, compared, out annotation), ifPart, elsePart, ifStatement.Condition);
+
+                case AssignmentExpressionSyntax assignment:
+                    var context = new ComparedContext(diagnostic, semanticModel, null, out annotation);
+                    if (assignment.Right is BinaryExpressionSyntax binaryExpression && binaryExpression.Kind() == SyntaxKind.CoalesceExpression)
+                    {
+                        return CoalesceAssignmentExpression(context, assignment.Left, binaryExpression.Right);
+                    }
+                    else if (assignment.Right is ConditionalExpressionSyntax conditional)
+                    {
+                        ConditionalSimplification.TryGetExpressionComparedToNull(conditional.Condition.RemoveParentheses(), out compared, out var comparedIsNullInTrue);
```

Comment:
To ensure the rule behaves consistently and as expected in all cases, the `RemoveParentheses()` method call should be done inside the `TryGetExpressionComparedToNull`. The following non-compliant code will have wrong message (asking to update to use the `?:` operator instead of asking user to use `??`)
```
            if ((a != null))
            {
                x = a;
            }
            else
            {
                x = b;
            }
```